rust/src/accumulator.rs

```diff
@@ -0,0 +1,224 @@
+//! This module contains the structure that accumulates usage data
+//! locally before flushing to Kafka.
+//!
+//! The accumulator pre-aggregates usage per timestamp based on
+//! the granularity provided at instantiation.
+//!
+
+use chrono::{DateTime, Duration, DurationRound, Local};
+use serde::Serialize;
+use std::collections::HashMap;
+use std::fmt;
+use std::mem;
+
+/// The unit of measures we support when recording usage.
+/// more can be added.
+#[derive(Clone, Debug, Eq, Hash, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum UsageUnit {
+    Milliseconds,
+    Bytes,
+    MillisecondsSec,
```

Comment:
How long is a "milliseconds sec" anyway?

Reply:
Three parsecs over 6 dollars square.

This was supposed to be bytes/sec, which is how GCP measures resources like network.
Will fix

Reply:
done